### PR TITLE
[consumer] Add new consumererrorprofiles module

### PR DESCRIPTION
**Pull request by mx-psi**
#### Description

Split off from #11131. Adds new module for consumer errors for profiles.

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Comment by mx-psi** [comment]
@open-telemetry/collector-approvers this is waiting on reviews!

**Review by andrzej-stencel — APPROVED** [review]

**Comment by dmathieu** [comment]
This seems ready to be merged?

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**
